### PR TITLE
Release process: Fix a confusing paragraph

### LINKED ISSUE #155: Rephrase the sentence in "Release process" regarding "compliant to yocto source"
In the document docs/chapters/workflow/release/release-process.rst we have a sentence which reads "The release process of PELUX sources is compliant to yocto source release which can be found on `Yocto Development Manual`_". This sentence is a bit confusing and it is not clear what is meant by it.
The word 'compliant' is confusing because that word is commonly used with licenses, like "compliant with the GPL" or similar.
What is meant is that if you follow the given link, there is a yocto guide on how to release and zip the sources.

### PR DESCRIPTION
**Pull request by thenor**
Also includes removal of unnecessary whitespace.

Fixes #155 

Signed-off-by: Therese Nordqvist <therese.nordqvist@pelagicore.com>

### CODE REVIEW

File: docs/chapters/workflow/release/release-process.rst [hunk 1]
```diff
@@ -15,7 +15,7 @@ Binary images (starting from Pelux 3.0) are now available for `download`_. Each
 
 Currently 4 different variants are released. Every hardware offers the qtauto image as well as the minimal image, while QEMU offers only the minimal image. 
 
-Besides the images, SDK is also released with the following format: 
+Besides the images, SDK is also released with the following format:
```
Comment (sashko) [inline docs/chapters/workflow/release/release-process.rst]: Still confusing. SDK installer which name has the following format or something like that?

Comment (thenor) [reply docs/chapters/workflow/release/release-process.rst]: Ok I can fix more stuff in the same file, but my focus was only to fix the paragraph referred to in the issue.

Comment (sashko) [reply docs/chapters/workflow/release/release-process.rst]: I just find the phrase `SDK is also released with the following format` confusing: it sounds like it's talking about the SDK format, when in fact, it describes installer's name format.

Comment (thenor) [reply docs/chapters/workflow/release/release-process.rst]: I tried to rephrase it. I hope it's good now.

**Review by sashko — APPROVED** [review]

**Review by JEderonn — APPROVED** [review]

**Pull request merged**